File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {}

DnaTranscriber.prototype.toRna = function(input) {
    let transcribed = '';
    for (let c of input) {
        transcribed += transcribe(c);
    }
    return transcribed;
}

var transcribe = function(letter) {
    switch (letter) {
        case 'G':
            return 'C';
        case 'C':
            return 'G';
        case 'T':
            return 'A';
        case 'A':
            return 'U';
        default:
            throw new Error('Invalid input');
    }
}

module.exports = DnaTranscriber;

File: java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription {
    String transcribe(String dnaStrand) {
    	String invalids = dnaStrand
    			.replace("G", "")
    			.replace("C", "")
    			.replace("A", "")
    			.replace("T", "");
    	if (invalids.length() > 0) {
    		throw new IllegalArgumentException("Invalid input");
    	}
        return dnaStrand
        		.replace('G', 'N')
        		.replace('C', 'G')
        		.replace('N', 'C')
        		.replace('A', 'U')
        		.replace('T', 'A');
    }
}